jsr311.go

```diff
@@ -41,9 +41,29 @@ func (r RouterJSR311) SelectRoute(
 
 // http://jsr311.java.net/nonav/releases/1.1/spec/spec3.html#x3-360003.7.2
 func (r RouterJSR311) detectRoute(routes []Route, httpRequest *http.Request) (*Route, error) {
+	ifOk := []Route{}
```

Comment:
Did you consider combining these two loops over the routes?
Looks like the method detection is independent of the IfFunction.

Reply:
I noticed that you've got a few passes in this method that iteratively filter through the routes and produce `methodOk`, `mediaOk`, and `inputMediaOk`, so I just followed that pattern and didn't consider something else.

Reply:
yes, ok leave it like this. maybe we could optimize later.  maybe put de method check (is cheap) before the conditional function loop.